src/main/java/ch/heigvd/statique/utils/Watcher.java

```diff
@@ -0,0 +1,142 @@
+     *
+     * @param dir directory to watch
+     */
+    public Watcher(Path dir) throws IOException {
+        watcher = FileSystems.getDefault().newWatchService();
+        this.dir = dir;
+    }
+
+    /**
+     * from : https://www.logicbig.com/tutorials/core-java-tutorial/java-nio/java-watch-service.html
+     *
+     * @param path directory path
+     */
+    private void registerDir(Path path) throws
+            IOException {
+        if (!Files.isDirectory(path, LinkOption.NOFOLLOW_LINKS) || path.getFileName().toString().equals("build")) {
+            return;
+        }
+
+        System.out.println("Adding " + path.getFileName() + " folder to watch");
+
+        WatchKey key = path.register(
+                watcher,
+                StandardWatchEventKinds.ENTRY_CREATE,
+                StandardWatchEventKinds.ENTRY_MODIFY,
+                StandardWatchEventKinds.ENTRY_DELETE
+        );
+        keyPathMap.put(key, path);
+
+
+        for (File f : Objects.requireNonNull(path.toFile().listFiles())) {
+            registerDir(f.toPath());
+        }
+    }
+
+    @Override
+    public void run() {
+        // from : https://docs.oracle.com/javase/tutorial/essential/io/notification.html
+        System.out.println("Watching...");
+        try {
+            registerDir(dir);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return;
+        }
+
+        while (true) {
+            // Wait for key to be signaled
+            WatchKey key;
+            try {
+                key = watcher.take();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+                break;
+            }
+            if (key == null)
+                continue;
+
+            for (WatchEvent<?> event : key.pollEvents()) {
+                WatchEvent.Kind<?> kind = event.kind();
+
+                // An OVERFLOW event can occur regardless if events are lost or discarded.
+                if (kind == OVERFLOW) {
+                    continue;
+                }
+
+
+                // The filename is the context of the event.
+                WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                Path filename = ev.context();
+
+                // Ignores build directory
+                if (filename.toString().equals("build")) {
+                    continue;
+                }
+
+
+                if (kind == ENTRY_CREATE) {
+                    //this is not a complete path
+                    Path path = ev.context();
+                    //need to get parent path
+                    Path parentPath = keyPathMap.get(key);
+                    //get complete path
+                    path = parentPath.resolve(path);
+
+                    try {
+                        registerDir(path);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+
+                if (ENTRY_CREATE.equals(kind)) {
```

Comment:
Couldn't you merge those two if together ?

```suggestion
                if (ENTRY_CREATE.equals(kind)) {
                    //this is not a complete path
                    Path path = ev.context();
                    //need to get parent path
                    Path parentPath = keyPathMap.get(key);
                    //get complete path
                    path = parentPath.resolve(path);

                    try {
                        registerDir(path);
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
```

Reply:
Yes, you're right ! Thx ^^